pkg/executor/join/hash_join_spill_helper.go

```diff
@@ -81,6 +81,7 @@ type hashJoinSpillHelper struct {
 	spillTriggedInBuildingStageForTest           bool
 	spillTriggeredBeforeBuildingHashTableForTest bool
 	allPartitionsSpilledForTest                  bool
+	skipProbeInRestoreForTest                    bool
```

Comment:
Why we need this?

Reply:
> Why we need this?

I find that one test case is not covered before and add it in this pr.